src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/GetCohortSummary/GetCohortSummaryQueryHandler.cs

```diff
@@ -24,59 +24,85 @@ public async Task<GetCohortSummaryQueryResult> Handle(GetCohortSummaryQuery requ
+                TransferSenderName = c.TransferSender.Name,
+                PledgeApplicationId = c.PledgeApplicationId,
+                WithParty = c.WithParty,
+                LatestMessageCreatedByEmployer = c.Messages.OrderByDescending(m => m.CreatedDateTime).Where(m => m.CreatedBy == 0).Select(m => m.Text).FirstOrDefault(),
+                LatestMessageCreatedByProvider = c.Messages.OrderByDescending(m => m.CreatedDateTime).Where(m => m.CreatedBy == 1).Select(m => m.Text).FirstOrDefault(),
+                ProviderId = c.ProviderId,
+                LastAction = c.LastAction,
+                LastUpdatedByEmployerEmail = c.LastUpdatedByEmployerEmail,
+                LastUpdatedByProviderEmail = c.LastUpdatedByProviderEmail,
+                Approvals = c.Approvals,
+                IsApprovedByEmployer = c.Approvals.HasFlag(Party.Employer), //redundant
+                IsApprovedByProvider = c.Approvals.HasFlag(Party.Provider), //redundant
+                IsCompleteForEmployer = CalculateIsCompleteForEmployer(c, apprenticeEmailIsRequired),
+                IsCompleteForProvider = CalculateIsCompleteForProvider(c, apprenticeEmailIsRequired),
+                LevyStatus = c.AccountLegalEntity.Account.LevyStatus,
+                ChangeOfPartyRequestId = c.ChangeOfPartyRequestId,
+                TransferApprovalStatus = c.TransferApprovalStatus,
+                ApprenticeEmailIsRequired = apprenticeEmailIsRequired
+            })
                 .SingleOrDefaultAsync(c => c.CohortId == request.CohortId, cancellationToken);
 
             return result;
         }
+
+        private static bool CalculateIsCompleteForProvider(Models.Cohort c, bool apprenticeEmailIsRequired)
+        {
+            return CalculateIsCompleteForEmployer(c, apprenticeEmailIsRequired)
+                && !c.Apprenticeships.Any(a => a.Uln == null);
+        }
+
+        private static bool CalculateIsCompleteForEmployer(Models.Cohort c, bool apprenticeEmailIsRequired)
+        {
+            return c.Apprenticeships.Any() && !c.Apprenticeships.Any(HasMissingData);
+
+            bool HasMissingData(Models.ApprenticeshipBase a)
+            {
+                if(a.FirstName == null
+                    || a.LastName == null
+                    || a.DateOfBirth == null
+                    || a.CourseName == null
+                    || a.StartDate == null
+                    || a.EndDate == null
+                    || a.Cost == null)
+                {
+                    return true;
+                }
+
+                if(apprenticeEmailIsRequired && a.Email == null && a.ContinuationOfId == null)
+                {
+                    return true;
+                }
+
+                if(a.DeliveryModel == DeliveryModel.PortableFlexiJob
+                    && (a.FlexibleEmployment?.EmploymentEndDate == null
+                    || a.FlexibleEmployment?.EmploymentPrice == null))
+                {
+                    return true;
+                }
```

Comment:
I like this, much better! However I don't think the Employer approval now gets affected by Flexi jobs, 

For flexi jobs, the employer is not entering EndDate or Cost, and therefore can they approve on their side even if these aren't present! This may be a BA question. We need to discuss with Teju